### ISSUE TITLE
Simple unreachable code cases not detected

### ISSUE THREAD

**Issue by 2-5** [microsoft/pyright#3635]
**Describe the bug**
Some trivial cases of unreachable code are not detected.

**To Reproduce**
Static `0`, `""`, `[]`, `{}` conditions are not detected as unreachable, unlike `False` which is.

**Expected behavior**
Trivial cases should be detected as unreachable. In particular `if 0:` which I encountered in the wild.

**Screenshots or Code**
```python
// NOT detected
if 0:
    print("unreachable")

// NOT detected
if "":
    print("unreachable")

// NOT detected
if []:
    print("unreachable")

// NOT detected
if {}:
    print("unreachable")

// detected
if False:
    print("unreachable")
```

**VS Code extension or command-line**
Pylance v2022.6.30
